src/rtk/features/singleRights/singleRightsSlice.ts

```diff
@@ -72,34 +71,43 @@ const singleRightSlice = createSlice({
   name: 'singleRightsSlice',
   initialState,
   reducers: {
-    removeServiceResource: (state: chosenServices, action) => {
-      state.chosenServices = state.chosenServices.filter(
+    removeServiceResource: (state: sliceState, action) => {
+      state.chosenServiceList = state.chosenServiceList.filter(
         (s) => s.service?.identifier !== action.payload,
       );
     },
   },
   extraReducers: (builder) => {
     builder.addCase(delegationAccessCheck.fulfilled, (state, action) => {
-      let chosenService: chosenService = {};
-      const delegableService = action.payload.find(
-        (response: delegationAccessCheckResponse) => response.status === 'Delegable',
+      const chosenService: ChosenService = {
+        accessCheckResponses: action.payload,
+        service: action.meta.arg.serviceResource,
+        status: 'Delegable',
+        errorCode: '',
+      };
+
+      const isNotDelegable = action.payload.find(
+        (response: delegationAccessCheckResponse) => response.status === 'NotDelegable',
       );
-      if (delegableService) {
-        chosenService = {
-          accessCheckResponses: action.payload,
-          service: action.meta.arg.serviceResource,
-          status: 'Delegable',
-        };
-      } else {
-        chosenService = {
-          accessCheckResponses: action.payload,
-          service: action.meta.arg.serviceResource,
-          status: 'NotDelegable',
-          errorCode: action.payload[0].details[0].code,
-        };
+
+      if (isNotDelegable) {
+        const isDelegable = action.payload.find(
```

Comment:
```suggestion
      if (hasNondelegableRights) {
        const hasDelegableRights= !!action.payload.find(
```
Consider changing  these vars to booleans and their names to something indicating that they have delegable/nondelegable rights, so that they aren't mutually exclusive.
It would probably be more readable considering this code checks if they're both true in order to assign the PartiallyDelegable status

Reply:
The !! operator is used to convert the function to a boolean but I'll modify code of hasDelegableRights to specifically say it's a boolean as well to make it more clear

Reply:
I fix